apps/ledger-live-mobile/src/screens/WalletCentricAsset/ReadOnly/index.tsx

```diff
@@ -42,8 +44,16 @@ const accounts: AccountLike[] = [];
 
 const ReadOnlyAssetScreen = ({ route }: NavigationProps) => {
   const { t } = useTranslation();
-  const currency = route?.params?.currency;
   const { colors } = useTheme();
+  const { currency: currencyParam, currencyId } = route?.params;
+  const marketParams = useSelector(marketParamsSelector);
+  const { counterCurrency = "usd" } = marketParams;
+
+  const { data: currencyData } = useCurrencyData(
+    currencyId ? { id: currencyId, counterCurrency } : { id: "", counterCurrency },
```

Comment:
Just for readability:
```suggestion
    { id: currencyId ?? "", counterCurrency },
```